src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
+
+  static removeDeadAnimal() {
+    Animal.alive = Animal.alive.filter((item) => item.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, hidden = false) {
+    super(name);
+    super.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (prey instanceof Herbivore && !prey.hidden && prey.health > 0) {
+      prey.health -= 50;
+
+      if (prey.health <= 0) {
+        Animal.removeDeadAnimal();
+      }
+    }
+  }
 }
 
+module.exports = { Animal, Herbivore, Carnivore };
+
 module.exports = {
```

Comment:
[CODE QUALITY] The `module.exports` is duplicated here. You should remove the second `module.exports` block (lines 42-46) to clean up the code and avoid potential confusion or errors.